hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/listener/ClientListenerServiceImpl.java

```diff
@@ -102,7 +107,9 @@ public UUID registerListener(final ListenerMessageCodec codec, final EventHandle
 
             ClientListenerRegistration registration = new ClientListenerRegistration(handler, codec);
             registrations.put(userRegistrationId, registration);
-            Collection<ClientConnection> connections = clientConnectionManager.getActiveConnections();
+            Collection<ClientConnection> connections = id == null
+                    ? clientConnectionManager.getActiveConnections()
```

Comment:
When a new connection is added, we add all the previously registered listeners to it. So, even if you do your registration to a single connection here, we will possibly try to register it to another connection